hare/algorithm.go

```diff
@@ -152,10 +152,14 @@ type ConsensusProcess struct {
 	cfg               config.Config
 	pending           map[string]*Msg // buffer for early messages that are pending process
 	notifySent        bool            // flag to set in case a notification had already been sent by this instance
+	mTracker          *msgsTracker    // tracks valid messages
 }
 
 // NewConsensusProcess creates a new consensus process instance.
-func NewConsensusProcess(cfg config.Config, instanceId instanceId, s *Set, oracle Rolacle, stateQuerier StateQuerier, layersPerEpoch uint16, signing Signer, nid types.NodeId, p2p NetworkService, terminationReport chan TerminationOutput, logger log.Log) *ConsensusProcess {
+func NewConsensusProcess(cfg config.Config, instanceId instanceId, s *Set, oracle Rolacle, stateQuerier StateQuerier,
```

Comment:
note to the future. it seems that you have a lot of constructor configurations. I'd pass a single `ConsensusProcesesConfig` struct that holds all the information needed to start a process.

Reply:
Ok, but do you mean a config that is a collection of all primitives? For example, a set is not a primitive... or do you mean just a single config that includes all?